lib/cartopy/io/img_tiles.py

```diff
@@ -48,22 +49,39 @@ class GoogleWTS(six.with_metaclass(ABCMeta, object)):
     A "tile" in this class refers to the coordinates (x, y, z).
 
     """
+    _MAX_THREADS = 24
+
     def __init__(self, desired_tile_form='RGB'):
         self.imgs = []
         self.crs = ccrs.Mercator.GOOGLE
         self.desired_tile_form = desired_tile_form
 
     def image_for_domain(self, target_domain, target_z):
         tiles = []
-        for tile in self.find_images(target_domain, target_z):
+
+        def fetch_tile(tile):
             try:
                 img, extent, origin = self.get_image(tile)
             except IOError:
-                continue
+                # Some services 404 for tiles that aren't supposed to be
+                # there (e.g. out of range).
+                raise
             img = np.array(img)
             x = np.linspace(extent[0], extent[1], img.shape[1])
             y = np.linspace(extent[2], extent[3], img.shape[0])
-            tiles.append([img, x, y, origin])
+            return img, x, y, origin
+
+        with concurrent.futures.ThreadPoolExecutor(
+                max_workers=self._MAX_THREADS) as executor:
+            futures = []
+            for tile in self.find_images(target_domain, target_z):
+                futures.append(executor.submit(fetch_tile, tile))
+            for future in concurrent.futures.as_completed(futures):
```

Comment:
Do we need to worry about order? Are they stitched together before drawing, or left to Matplotlib? If the latter, they should be sorted to get consistent draws.

Reply:
So matplotlib is an implementation detail that we shouldn't need to care about in this module. As it happens in this case we call ``_merge_tiles`` which brings together all tiles (in any order) into a single image. So the simple answer is - no worries about order.

What is interesting is that I have a prototype (developed yesterday) that passes the raw tiles through to matplotlib for it to merge together (in the appropriate backend in the form of a rendered figure) and in that case the order of the tiles doesn't matter either.

Long and short - order is no concern. 👍

Reply:
> prototype

Oh, will this be fixing the PDF rasterization bug?